config/main.py

```diff
@@ -117,6 +132,140 @@ def get_command(self, ctx, cmd_name):
 except KeyError, TypeError:
     raise click.Abort()
 
+#
+# Breakout Mode Helper functions
+#
+
+# Read given JSON file
+def readJsonFile(fileName):
+    try:
+        with open(fileName) as f:
+            result = json.load(f)
+    except Exception as e:
+        raise Exception(str(e))
+    return result
+
+def _get_option(ctx,args,incomplete):
+    """ Provides dynamic mode option as per user argument i.e. interface name """
+    global all_mode_options
+    interface_name = args[-1]
+
+    if not os.path.isfile(BREAKOUT_CFG_FILE) or not BREAKOUT_CFG_FILE.endswith('.json'):
+        return []
+    else:
+        breakout_file_input = readJsonFile(BREAKOUT_CFG_FILE)
+        if interface_name in breakout_file_input[INTF_KEY]:
+            breakout_mode_list = [v["breakout_modes"] for i ,v in breakout_file_input[INTF_KEY].items() if i == interface_name][0]
+            breakout_mode_options = []
+            for i in breakout_mode_list.split(','):
+                    breakout_mode_options.append(i)
+            all_mode_options = [str(c) for c in breakout_mode_options if incomplete in c]
+            return all_mode_options
+
+
+def shutdown_interfaces(ctx, del_intf_dict):
+    """ shut down all the interfaces before deletion """
+    for intf in del_intf_dict.keys():
+        config_db = ctx.obj['config_db']
+        if get_interface_naming_mode() == "alias":
+            interface_name = interface_alias_to_name(intf)
+            if interface_name is None:
+                click.echo("[ERROR] interface name is None!")
+                return False
+        if interface_name_is_valid(intf) is False:
+            click.echo("[ERROR] Interface name is invalid. Please enter a valid interface name!!")
+            return False
+        if intf.startswith(PORT_STR):
```

Comment:
This assumes front-panel ports in SONiC will always start with "Ethernet". If we decide to change the port naming convention, this will need to be reworked. Is there a more generic way to approach this?

Reply:
I guess we are moving this whole function to config_mgmt.py as we want this function to be part of config mgmt [API.](https://github.com/Azure/sonic-utilities/pull/765/files) Right @praveen-li?

Now regarding your feedback, One approach we can take is like below.

 ```
port_dict = config_db.get_table('PORT')
if not port_dict:
    click.echo("port_dict is None!")
    raise click.Abort()
 if intf in port_dict.keys():
     config_db.mod_entry("PORT", intf, {"admin_status": "down"})
```

How about it? @jleveque, kindly suggest. @praveen-li, should I update the function here or will you move the function in "config_mgmt.py" file any time soon?

Reply:
I think that approach should work, and not be dependent on the naming conventions.

Reply:
Modified as requested.